CONTRIBUTING.md

```diff
@@ -1,162 +1,152 @@
+to work on, and why. The Crossplane maintainers can't always triage new issues
+immediately, but we encourage you to bring them to our attention via [Slack].
+
+Be sure to practice [good git commit hygiene] as you make your changes. All but
+the smallest changes should be broken up into a few commits that tell a story.
+Use your git commits to provide context for the folks who will review PR, and
+the folks who will be spelunking the codebase in the months and years to come.
+Ensure each of your commits is signed-off in compliance with the [Developer
+Certificate of Origin] by using `git commit -s`. The Crossplane highly values
+readable, idiomatic Go code. Familiarise yourself with common [code review
+comments] that are left on Go PRs - try to preempt any that your reviewers would
+otherwise leave. Run `make reviewable` to lint your change.
+
+All Crossplane code must be covered by tests. Note that unlike many Kubernetes
+projects Crossplane does not use gingko tests and will request changes to any PR
+that uses gingko or any third party testing library, per the common Go [test
+review comments]. Crossplane encourages the use of table driven unit tests. The
+tests of the [crossplane-runtime] project are representative of the testing
+style Crossplane encourages; new tests should follow their conventions. Note
+that when opening a PR your reviewer will expect you to detail how you've tested
+your work. For all but the smallest changes some manual testing is encouraged in
+addition to unit tests.
+
+All Crossplane documentation and examples are under revision control; see the
+[docs] and [examples] directories of this repository. Any change that introduces
+new behaviour or changes existing behaviour must include updates to any relevant
+documentation and examples. Please keep documentation and example changes in
+distinct commits.
+
+One your change is written, tested, and documented the final step is to have it
+reviewed! You'll be presented with a template and a small checklist when you
+open a PR. Please read the template and fill out the checklist. If the folks
+reviewing your PR request changes please make them in new commits. This allows
+your reviewers to see what has changed as you address their comments. Be mindful
+of  your commit history as you do this - avoid commit messages like "Address
+review feedback" if possible. If doing so is difficult a good alternative is to
+rewrite your commit history to clean them up after your PR is approved but
+before it is merged.
+
+In summary, please:
+
+* Discuss your change in a GitHub issue before you start.
+* Use your Git commit messages to communicate your intent to your reviewers.
+* Sign-off on all Git commits by running `git commit -s`
+* Add or update tests for all changes.
+* Preempt common [code review comments] and [test review comments].
+* Update all relevant documentation and examples.
+* Don't force push to address review feedback.
```

Comment:
want to add another item in this list about cleaning up (squashing) PR feedback commits before merging?

Reply:
Sure.